redundant-bmc/src/services_impl.cpp

```diff
@@ -65,6 +65,86 @@ sdbusplus::async::task<std::string> getService(sdbusplus::async::context& ctx,
     co_return object.begin()->first;
 }
 
+/**
+ * @brief Run a shell command asynchronously
+ *
+ * Use the pipe()/fork() paradigm to fork off a child process to run
+ * the command. The child will write the command RC to its FD obtained
+ * from pipe() and exit. Meanwhile the parent will use async::fdio to
+ * asynchronously wait for that rc to show up in its read FD it had
+ * obtained from pipe().
+ */
+// NOLINTNEXTLINE
+sdbusplus::async::task<int> runAsyncCmd(sdbusplus::async::context& ctx,
+                                        const std::string& cmd)
+{
+    int pipeFDs[2];
+
+    // Open the read and write pipes
+    if (pipe(pipeFDs) == -1)
+    {
+        auto e = errno;
+        lg2::error("runAsyncCmd: pipe() failed with errno: {ERRNO}", "ERRNO",
+                   e);
+        co_return -1;
+    }
+
+    pid_t pid = fork();
+    if (pid == -1)
+    {
+        auto e = errno;
+        close(pipeFDs[0]);
+        close(pipeFDs[1]);
+        lg2::error("runAsyncCmd: fork failed with errno {ERRNO}", "ERRNO", e);
+        co_return -1;
+    }
+    else if (pid == 0)
+    {
+        // Child
+
+        // Close the read pipe
+        close(pipeFDs[0]);
+
+        // NOLINTNEXTLINE(cert-env33-c)
+        int rc = std::system(cmd.c_str());
+
+        // Write the return code to the write pipe
+        ssize_t s = write(pipeFDs[1], &rc, sizeof(rc));
```

Comment:
If the intent is to capture the actual exit status of `journalctl --sync` for debugging, relying on the return value from `std::system` is insufficient. The correct approach is to extract the exit code via `WEXITSTATUS`, for example:

```
int rc = std::system(cmd.c_str());
int exitCode = (rc == -1) ? -1 :
               (WIFEXITED(rc) ? WEXITSTATUS(rc) : -1);
write(pipeFDs[1], &exitCode, sizeof(exitCode));
```

Additionally, this results in two forks: one explicitly at line 92 and another internally from `std::system`. Given the available 10-second window, the extra overhead should not pose a performance concern.